File: Week1/Project1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int coinCalculation(float change);

int main(void)
{
    float change;

    // Prompt for users change owed.
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    // The result of the calculation is stored in the local 'coins' variable.
    int coins = coinCalculation(change);

    // Prints total number of coins as an integer.
    printf("%i\n", coins);
}

int coinCalculation(float change)
{

    // Converts change to cents.
    int cents = round(change);
    // round() from the math.h library, rounds the value to the nearest integer.

    // Declaration of coins values.
    int quarters = 25;
    int dimes = 10;
    int nickels = 5;
    int pennies = 1;

    int coins = 0;

    // Coins calculation per coin.
    coins += cents / quarters;
    cents = cents % quarters;

    coins += cents / dimes;
    cents = cents % dimes;

    coins += cents / nickels;
    cents = cents % nickels;

    coins += cents / pennies;
    cents = cents % pennies;

    // Return the final count to main.
    return coins;
}

File: Week6/Pset6/Cash/sentimental-cash/cash.py
from cs50 import get_float

# Prompt users change owed.
while True:
    change = get_float("Change owed: ")
    if change >= 0:
        break

# Convert change to cents.
cents = round(change * 100)

coins = 0

# Quarters.
coins += cents // 25
cents %= 25

# Dimes.
coins += cents // 10
cents %= 10

# Nickels.
coins += cents // 5
cents %= 5

# Pennies.
coins += cents // 1
cents %= 1

# Print the minimum number of coins.
print(coins)
